src/psyclone/tests/domain/nemo/transformations/acc_update_test.py

```diff
@@ -0,0 +1,456 @@
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors: R. W. Ford, A. R. Porter, S. Siso and N. Nobre, STFC Daresbury Lab
+
+'''Module containing py.test tests for the transformation of the PSy
+   representation of NEMO code using OpenACC update directives.
+
+'''
+
+import pytest
+
+from fparser.common.readfortran import FortranStringReader
+from fparser.two import Fortran2003
+
+from psyclone.psyGen import PSyFactory
+from psyclone.psyir.nodes import ACCKernelsDirective, CodeBlock, Schedule
+from psyclone.psyir.transformations import TransformationError, ACCUpdateTrans
+from psyclone.transformations import ACCEnterDataTrans, ACCKernelsTrans
+from psyclone.tests.utilities import Compile
+
+# Constants
+API = "nemo"
+
+
+def test_validate():
+    ''' Check that the validate() method works as expected. '''
+    trans = ACCUpdateTrans()
+
+    with pytest.raises(TransformationError) as err:
+        trans.validate(ACCKernelsDirective())
+    assert ("Expected a Schedule but got a node of type 'ACCKernelsDirective'"
+            in str(err.value))
+
+    with pytest.raises(TransformationError) as err:
+        trans.validate(Schedule(parent=ACCKernelsDirective()))
+    assert ("Cannot apply the ACCUpdateTrans to nodes that are "
+            "within an OpenACC compute region." in str(err.value))
+
+
+def test_simple_missed_region(parser):
+    ''' Check code generation when there is a simple section of host code
+    between two kernels regions. '''
+    code = '''
+SUBROUTINE tra_ldf_iso()
+  USE some_mod, only: dia_ptr_hst
+  INTEGER, PARAMETER :: jpi=2, jpj=2, jpk=2
+  INTEGER :: jn
+  LOGICAL :: l_ptr
+  INTEGER, DIMENSION(jpi,jpj) :: tmask
+  REAL, DIMENSION(jpi,jpj,jpk) ::   zdit, zdjt, zftu, zftv, ztfw
+  zftv(:,:,:) = 0.0d0
+  IF( l_ptr )THEN
+    CALL dia_ptr_hst( jn, 'ldf', zftv(:,:,:)  )
+    zftv(:,:,:) = 1.0d0
+  END IF
+  CALL dia_ptr_hst( jn, 'ldf', zftv(:,:,:)  )
+  zftu(:,:,1) = 1.0d0
+  tmask(:,:) = jpi
+end SUBROUTINE tra_ldf_iso
+'''
+    reader = FortranStringReader(code)
+    ast = parser(reader)
+    psy = PSyFactory(API, distributed_memory=False).create(ast)
+    schedule = psy.invokes.invoke_list[0].schedule
+    acc_trans = ACCEnterDataTrans()
+    acc_kernels = ACCKernelsTrans()
+    acc_update = ACCUpdateTrans()
+    # We do not permit arbitrary code blocks to be included in data
+    # regions so just put three of the (implicit) loops into kernels.
+    acc_kernels.apply(schedule[0])
+    acc_kernels.apply(schedule[3:5])
+    acc_trans.apply(schedule)
+    acc_update.apply(schedule)
+    gen_code = str(psy.gen).lower()
```

Comment:
Please use the fortran_writer fixture instead of the psy.gen in all this tests.

Reply:
Done.